packages/chat-sdk/src/channel/channel.ts

```diff
@@ -983,27 +985,122 @@ export class Channel implements BaseChannel {
+    }
+
+    if (callback && type) {
+      if (this.chatConfigChangesCallbacks[type].indexOf(callback)) {
+        const newCallbacksList = this.chatConfigChangesCallbacks[type].filter((item) => item !== callback);
+        this.chatConfigChangesCallbacks[type] = newCallbacksList;
+      }
+    }
+  }
+
+
   /**
    * Watch chat config changes
    *
    */
-  public watchChatConfigChanges(callback?: (channel: LiveChatChannel) => void): () => void {
+  public watchChatConfigChanges(callback?: (item: any) => void, type?: ChatConfigType): () => void {
     try {
-      const realtimeAPI = RealtimeAPI.getInstance();
-      const path = `/chat-rooms/${this.chatRoom._id}/channels/${this.channel._id}`;
-      return realtimeAPI.listenToChatConfigChanges(path, (nextChatRoom) => {
-        this.channel = {
-          ...nextChatRoom,
-          _id: this.channel._id,
-          dataPath: this.channel.dataPath,
-        };
-
-        if (callback) {
-          callback(this.channel);
+      if (callback) {
+        const callbackType = type || ChatConfigType.ALL_CHAT_CHANGES;
+
+        if (this.chatConfigChangesCallbacks[callbackType]) {
+          this.chatConfigChangesCallbacks[callbackType].push(callback);
+        } else {
+          this.chatConfigChangesCallbacks[callbackType] = [callback];
         }
-      });
+      }
+
+      if (!this.chatConfigListenerUnsubscribe) {
+        const realtimeAPI = RealtimeAPI.getInstance();
+        const path = `/chat-rooms/${this.chatRoom._id}/channels/${this.channel._id}`;
+
+        this.chatConfigListenerUnsubscribe = realtimeAPI.listenToChatConfigChanges(path, (nextChatRoom) => {
+          this.onChatChanges(nextChatRoom);
+
+          if (this.pinnedMessageId !== nextChatRoom.pinnedMessageId) {
+            const newPinnedMessageId = nextChatRoom.pinnedMessageId || null;
+
+            this.pinnedMessageId = newPinnedMessageId;
+            this.onPinMessageChanges(newPinnedMessageId);
+          }
+        });
+      }
+
+      return () => {
+        this.offChatConfigListener(callback, type);
+      };
     } catch (e) {
       throw new Error('Cannot listen to chat config changes');
     }
   }
+
+
+  /**
+   * Remove all chat config listeners
+   *
+   */
+   public offAllChatConfigListeners(): void {
+    try {
+      if (this.chatConfigListenerUnsubscribe) {
+        this.chatConfigListenerUnsubscribe();
+        this.chatConfigListenerUnsubscribe = null;
+      }
+
+      this.chatConfigChangesCallbacks[ChatConfigType.ALL_CHAT_CHANGES] = [];
+      this.chatConfigChangesCallbacks[ChatConfigType.PIN_MESSAGES] = [];
+
+    } catch (err) {
+      throw new Error('Cannot off chat config changes');
+    }
+  }
+
+  /**
+   * Watch new config changes on chat
+   *
+   */
+  private onChatChanges(nextChatRoom: LiveChatChannel): void {
+    this.channel = {
+      ...nextChatRoom,
+      _id: this.channel._id,
+      dataPath: this.channel.dataPath,
+    };
```

Comment:
We can move this assign to the callback that we will implement to handle the `this.watchChatConfigChanges` defined in the constructor.